frontend/console/src/components/List.tsx

```diff
@@ -0,0 +1,24 @@
+import { classNames } from '../utils'
+
+type ListProps<T> = {
+  items: T[]
+  renderItem: (item: T) => React.ReactNode
+  onClick?: (item: T) => void
+  className?: string
+}
+
+export const List = <T,>({ items, renderItem, onClick, className }: ListProps<T>) => {
+  return (
+    <ul className={classNames('divide-y divide-gray-100 dark:divide-gray-700 overflow-hidden', className)}>
+      {items.map((item, index) => (
+        <li
+          key={index}
+          className={`relative flex justify-between gap-x-6 p-4 ${onClick ? 'cursor-pointer hover:bg-gray-50 dark:hover:bg-gray-700' : ''}`}
+          onClick={onClick ? () => onClick(item) : undefined}
```

Comment:
Just to confirm, if you use this component without an `onClick`, does passing `undefined` into this prop definitely not throw a runtime error? I wasn't sure if it was safe to pass `undefined` to a `fn` prop.

Reply:
yeah should be ok. All the current stuff is using it without an `onClick`.